File: sangsoo/0x15 Hash/1620.py
n, m = map(int, input().split())
si_dict = {}
is_dict = {}
idx = 1

def isNumber(s : str) -> bool:
    for i in range(len(s)):
        if not (48 <= ord(s[i]) <= 57): # 아스키 코드상 숫자가 아니면
            return False
    return True

for i in range(n):
    name = input()
    si_dict[name] = idx
    is_dict[idx] = name
    idx += 1
for i in range(m):
    in_str = input()
    if (isNumber(in_str)):
        num = int(in_str)
        print(is_dict[num])
    else:
        print(si_dict[in_str])

File: sangsoo/0x15 Hash/1620.cpp
#include <iostream>
#include <unordered_map>
#include <string>

using namespace std;
int N, M;

bool isNumber(const string &str){
    for (char  const &c : str){
        if (isdigit(c) == 0) return false;
    }
    return true;
}

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);
    cin >> N >> M;
    unordered_map<string, int> si_m; // stoi 로 명명하면 stoi 함수랑 겹침
    unordered_map<int, string> is_m;
    int idx = 1;
    for(int i = 0; i < N; i++){
        string name;
        cin >> name;
        si_m[name] = idx;
        is_m[idx] = name;
        idx += 1;  // 실수 주의
    }
    for (int i = 0; i < M; i++){
        string in;
        cin >> in;
        if (isNumber(in)) {
            int num = stoi(in);
            cout << is_m[num] << '\n';
        }
        else{
            cout << si_m[in] << '\n';
        }
    }
}